nixos/modules/config/defaults.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.users.defaults;
+in {
+  options.users.defaults = {
+    editor = mkOption {
+      type = types.editorPackage;
+      default = pkgs.nano;
+      defaultText = literalExample "pkgs.nano";
+      example = literalExample "pkgs.kakoune";
+      description = "The package to use as the default editor.";
+    };
+
+    editorCommand = mkOption {
+      type = types.str;
+      default = cfg.editor.editorCommand;
+      defaultText = literalExample "config.users.defaults.editor.editorCommand";
+      example = "gvim";
+      description = "The command to set $EDITOR to.";
+    };
```

Comment:
- If only `editorCommand` is set, `nano` will still be put on `PATH`.
 - The `editorCommand` example doesn't work without adding `gvim` to `PATH`.

Perhaps instead of `editorCommand` we could have `editorArguments`.
This doesn't preclude changing the executable, which can be done via `nano // { editorCommand = "rnano"; }` and still needs to be documented, but at least the relation between the `editor` and `editorArguments` options is clear at a glance.

Reply:
`editorCommand`/`pagerCommand` could also be dropped entirely

Reply:
This is a good point, and I feel it's somewhat unclean how `<thing>` and `<thing>Command` are not tied together in the module options; using the merge operator like that is potentially more clean. I like it!

Reply:
I removed these options and added documentation on how to change the commands otherwise.